src/applications/apps.js

```diff
@@ -283,7 +279,6 @@ export function validateRegisterWithConfig(config) {
           "The config.name on registerApplication must be a non-empty string"
       )
     );
-
   if (!config.app)
```

Comment:
This should check for function or plain object

Reply:
Also line 219 should do the same

Reply:
Promises are not valid as an `app` argument. It must be either a function or an object iwth the lifecycles on it.

Reply:
Should I check for lifeCycle keys in this object, or just plain object is enough?